src/hotspot/share/utilities/filterQueue.hpp

```diff
@@ -48,24 +52,32 @@ class FilterQueue {
  public:
   FilterQueue() : _first(NULL) { }
 
-  // MT-safe
   bool is_empty() {
     return load_first() == NULL;
   }
 
-  // MT-safe
-  void add(E data);
+  // Adds an item to the queue in a MT safe way, re-entrant.
+  void push(E data);
 
-  // MT-Unsafe, external serialization needed.
+  // Applies the match_func to the items in the queue until match_func returns
+  // true and then return false, or there is no more items and then returns
+  // false. Any pushed item while executing may or may not have match_func
```

Comment:
s/pushed item/item pushed/

I know what you are trying to say about concurrent pushes but it sounds too non-deterministic - any concurrent push that happens before contains() reaches the end of the queue will have the match_func applied. So in that sense "while executing" only applies to pushes that will be seen; any push not seen had to have happened after execution was complete.